Source/SwiftLintBuiltInRules/Rules/Style/StatementPositionRule.swift

```diff
@@ -36,175 +20,217 @@ struct StatementPositionRule: CorrectableRule {
-            Example("    }\n    else {"),
-            Example("  }\n  catch {"),
-            Example("  }\n\n  catch {"),
-            Example("\n\n  }\n  catch {"),
-            Example("\"}\nelse{\""),
-            Example("struct A { let catchphrase: Int }\nlet a = A(\n catchphrase: 0\n)"),
-            Example("struct A { let `catch`: Int }\nlet a = A(\n `catch`: 0\n)"),
-        ],
-        triggeringExamples: [
-            Example("↓  }else if {"),
-            Example("↓}\n  else {"),
-            Example("↓  }\ncatch {"),
-            Example("↓}\n\t  catch {"),
-        ],
-        corrections: [
-            Example("  }else if {"): Example("  }\n  else if {"),
-            Example("}\n  else {"): Example("}\nelse {"),
-            Example("  }\ncatch {"): Example("  }\n  catch {"),
-            Example("}\n\t  catch {"): Example("}\ncatch {"),
-        ]
+        nonTriggeringExamples: StatementPositionRuleExamples.uncuddledNonTriggeringExamples,
+        triggeringExamples: StatementPositionRuleExamples.uncuddledTriggeringExamples,
+        corrections: StatementPositionRuleExamples.uncuddledCorrections
     )
+}
 
-    func validate(file: SwiftLintFile) -> [StyleViolation] {
-        switch configuration.statementMode {
-        case .default:
-            return defaultValidate(file: file)
-        case .uncuddledElse:
-            return uncuddledValidate(file: file)
+// MARK: - Shared Validation Logic
+
+private struct StatementValidation {
+    let hasLeadingNewline: Bool
+    let hasTrailingContent: Bool
+    let expectedIndentation: Int
+    let actualIndentation: Int
+    let isSingleSpace: Bool
+    let hasCommentsBetween: Bool
+
+    init(keyword: TokenSyntax, previousToken: TokenSyntax) {
+        self.hasLeadingNewline = keyword.leadingTrivia.contains { piece in
+            switch piece {
+            case .newlines, .carriageReturns, .carriageReturnLineFeeds:
+                return true
+            default:
+                return false
+            }
         }
+        self.hasTrailingContent = !previousToken.trailingTrivia.isEmpty
+        self.expectedIndentation = Self.calculateIndentation(previousToken.leadingTrivia)
+        self.actualIndentation = Self.calculateIndentation(keyword.leadingTrivia)
+        self.isSingleSpace = previousToken.trailingTrivia.isSingleSpace
+
+        // Check for comments between closing brace and keyword
+        self.hasCommentsBetween = previousToken.trailingTrivia.contains(where: \.isComment) ||
+                                  keyword.leadingTrivia.contains(where: \.isComment)
     }
 
-    func correct(file: SwiftLintFile) -> Int {
-        switch configuration.statementMode {
-        case .default:
-            defaultCorrect(file: file)
-        case .uncuddledElse:
-            uncuddledCorrect(file: file)
+    private static func calculateIndentation(_ trivia: Trivia) -> Int {
+        var indentation = 0
+        for piece in trivia.reversed() {
```

Comment:
Why reversed?

Reply:
To calculate the indentation of the *last* line in the trivia, not the first. I'll add a comment.

Reply:
Without the `reversed()`, all tests still pass. So we should probably have a test case that makes sure reversed order is required.